src/kaltura-player.js

```diff
@@ -923,6 +930,19 @@ class KalturaPlayer extends FakeEventTarget {
     });
   }
 
+  _handleImageSrc(imgSrc: ImageSource): void {
+    let thumbnailAPIParams = {};
+    if (!this.config.session.isAnonymous) thumbnailAPIParams['ks'] = this.config.session.ks;
```

Comment:
note that we already have a handling for appending ks to thumbnails, and that it's toggled by a parameter in the provider:

https://github.com/kaltura/kaltura-player-js/blob/master/src/kaltura-player.js#L195

Reply:
this logic moved to the engine